xbmc/pvr/addons/PVRClient.h

```diff
@@ -169,6 +169,12 @@ class CPVRClient : public ADDON::IAddonInstanceHandler
    */
   const std::string GetFriendlyName() const;
 
+  /*!
+   * @brief The name used by the PVR client addon instance
+   * @return string that can be used in log messages and the GUI.
+   */
+  const std::string GetInstanceName() const;
```

Comment:
I don't think return const string is of any value. Return either const ref (if safe) or the string itself (copy).

Reply:
@enen92 - Thank you for your feedback.  I’m not trying to be argumentative, however, I would like to better understand your reasoning for requesting this change.

For consistency, I followed precedent and used `const std::string` because the new function `GetInstanceName` returns a subset of the data already returned by `GetFriendlyName` which also returns a `const std::string`.

What is the benefit to having these two very similar functions return their data differently?

Reply:
it's general rule of thumb to never return a `plain const` from any function.
though it may be done a lot across the codebase.

reasoning is that it is interfering with move semantics.. e.g.
```cpp
const std::string GetSomeString()
{
  std::string someString = "bla";
  return someString;
}

std::string anotherString = GetSomeString();

// this cannot move the string from the temporary into "anotherString"
// you cannot move from a const temporary. while it is possible to move from
// std::string GetSomeString()
```

so for new code you should return either `std::string` or `const std::string&`

Reply:
@howie-f - Thank you for your explanation, it was helpful.

I copied and modified the `GetFriendlyName` function to produce my new function.  If `GetFriendlyName` is implemented incorrectly, and it is the function immediately above my new function, should I change both functions to comply with the new formatting whilst I am working in that area of code?

Reply:
I used `std::string`.
I did not edit `GetFriendlyName` due to the lack of a response (Perfectly understandable considering the differing timezones involved).

Reply:
> If `GetFriendlyName` is implemented incorrectly...

i wouldn't call it implemented incorrectly because it is working. there's just a tiny room for optimization. personally i'd just leave `GetFriendlyName` as is like you've done now